### PR TITLE
[FIX] stock, purchase_stock, sale_stock: fix dark mode readability issues

### PR DESCRIPTION
**Pull request by chgo-odoo**
This PR fixes readability issues within the forecast details table.

Prior to this PR, the table was using arbitrary color that were out of our color maps, meaning that they were not dark mode proof by default.

As no variation was defined for these color in dark mode, some readability issues would occur in some cases (e.g link color over that background in dark mode).

To handle this issue, we go back to basics and use Bootstrap default table utility classes, easing the maintenance and preventing. readability issues.

| 16.0 | This PR |
|--------|--------|
| image | image | 

task-4344045

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by chgo-odoo** [comment]
@stefanorigano here is the fix, I switch the red color for the warning one, I believe this is what it should be according to the classname. To be checked with CRL or TIC in tech. review.

Could you have a look ?

Thanks

File: addons/stock/static/src/scss/stock_forecasted.scss [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
     .o_report_replenishment {
         .o_grid_warning {
-            background-color: #f4cccc;
+            --table-bg: #{map-get($table-variants, 'warning')};
         }
         .o_grid_match {
-            background-color: #f0f0f0;
+            --table-bg: #{map-get($table-variants, 'light')};
         }
     }
```
Comment (stefanorigano) [inline addons/stock/static/src/scss/stock_forecasted.scss:9]: Adding the `table-warning` and  `table-light` classes makes this entire block of CSS  redundant, we can remove it.

File: addons/stock/static/src/stock_forecasted/forecasted_details.xml [hunk 2]
```diff
@@ -23,8 +23,8 @@
                     <td/>
                     <td/>
                 </tr>
-                <tr t-foreach="props.docs.lines" t-as="line" t-key="line.index" t-attf-class="#{line.is_matched and 'o_grid_match'}">
-                    <td t-attf-class="#{line.is_late and 'o_grid_warning'}">
+                <tr t-foreach="props.docs.lines" t-as="line" t-key="line.index" t-attf-class="#{line.is_matched and 'o_grid_match table-light'}">
```
Comment (stefanorigano) [inline addons/stock/static/src/stock_forecasted/forecasted_details.xml:26]: I’d remove the custom classes; they seem unnecessary without their relative SCSS.

```suggestion
                <tr t-foreach="props.docs.lines" t-as="line" t-key="line.index" t-attf-class="#{line.is_matched and 'table-light'}">
```

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Thanks @chgo-odoo , some little remarks.
Once done you can move to tech. review directly.

>  I switched the red color for the warning one, I believe this is what it should be according to the classname.

Makes sense to me as well 👍

File: addons/stock/static/src/scss/stock_forecasted.scss (hunk 1 above)
Comment (chgo-odoo) [reply addons/stock/static/src/scss/stock_forecasted.scss:9]: Indeed, I thought it was needed but these templates aren't stocked within the DB so it won't be applied, thanks 😅

File: addons/stock/static/src/stock_forecasted/forecasted_details.xml (hunk 2 above)
Comment (stefanorigano) [reply addons/stock/static/src/stock_forecasted/forecasted_details.xml:26]: > someone could potentially target this class with an xpath ?

I find it very unlikely.
`o_grid_match` "it's not a class" just yet, so it cannot be targeted with `tr[hasclass('o_grid_match')]`, for example.
To target `//tr[last()]` or `//tr[@t-key="line.index"]` would be much more convenient than trying to match the entire  `t-attf-class` attribute 🤔

Anyhow, removing the classes it's not "urgent", these just don't do nothing.
If this change raises eyebrows during tech. review we can keep them in prod. and remove them in the fw port targeting `master`.

Don't resolve this conversation for now, let's let the devs decide ;)

**Review by stefanorigano — APPROVED** [review]

**Review by ticodoo — COMMENTED** [review]
The fix seems fine to me, my main concern is how different the colors are in not dark mode

currently:
image

vs:
image


Can we use `table-danger` and `table-info` instead? For these results:

image

image

This way the  `table-danger` = closer to the original color and  `table-info` = more visible than `table-light` (in dark more, `table-light` looks more like the background color is missing rather than the line is being emphasized imo)

image

**Comment by stefanorigano** [comment]
> my main concern is how different the colors are in not dark mode

@ticodoo ,  normally we use `warning` (~orange) to highlight situations that require attention but are not immediately critical, while `danger` (~red) to mark situations that could lead to significant harm, or data loss, or errors.

Still, I do agree that we shouldn't drastically change the current visual result in production otherwise people may get lost, you're right, good catch 👍 

@dalaOdoo , maybe we can can switch to `warning` (~orange) in the `master` fw port? What's your take?

**Comment by dalaOdoo** [comment]
@stefanorigano warning in master seems ok to me

**Comment by chgo-odoo** [comment]
Hello @ticodoo, could we move this forward as it is related to a customer ticket, please? 🙏 

Thanks !

**Comment by ticodoo** [comment]
Thanks!
@robodoo r+

**Pull request closed without merging**